c_tests/tests/noopts.c

```diff
@@ -0,0 +1,50 @@
+// Compiler:
+// Run-time:
+//   env-var: YKD_PRINT_IR=1
+//   stderr:
+//     ...
+//     store i32 2...
+//     ...
+//     store i32 3...
+//     ...
+//     ...add...
+//     ...
+
+// Check that our NOOPT macro blocks optimisations.
+
+#include <assert.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <yk_testing.h>
+
+int add_them(int x, int y) { return x + y; }
+
+int main(int argc, char **argv) {
+  int x, y, res;
+  void *tt = __yktrace_start_tracing(HW_TRACING, &res, &x, &y);
+  // The __yktrace_start_tracing() call has almost certainly already blocked
+  // optimisation of x and y, as pointers to them are passed as arguments and
+  // the compiler must assume that it may mutate them. Therefore we initialise
+  // x and y after the call.
+  x = 2;
+  NOOPT(x);
+  y = 3;
+  NOOPT(y);
+  // An optimising compiler would inline this call and (without our NOOPTs)
+  // would then use constant prorogation to replace the whole call with the
```

Comment:
I wasn't expecting parliament to make its effects felt here :) s/prorogation/propogation/

Reply:
Just you wait until you make a typo ;)